### PR TITLE
refactor: provider permissions and ens resolution

### LINKED ISSUE #3995: opening simpledapp.eth does not request the wallet
work on this is already ongoing but tracking here none the less: opening simpledapp.eth does not request the wallet, and wallet shows as already connected, the dapp buttons don’t work

### PR DESCRIPTION
**Pull request by richard-ramos**
Requires
- https://github.com/status-im/status-lib/pull/95

Fixes #3995

### CODE REVIEW

**Comment by richard-ramos** [comment]
Rebased

File: src/app_service/service/provider/service.nim [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    ensService: ens_service.ServiceInterface): Service =
+  result = Service()
+  result.dappPermissionsService = dappPermissionsService
+  result.settingsService = settingsService
+  result.ensService = ensService
+
+method init*(self: Service) =
+  discard
+
+proc process(self: Service, data: Web3SendAsyncReadOnly): string =
```
Comment (borismelnik) [inline src/app_service/service/provider/service.nim:111]: We have any chance that `data` can be a null?

**Review by iurimatias — CHANGES_REQUESTED** [review]
I tried a dapp and see an issue that does not happen in 0.3.0

see https://www.sorbet.finance/#/limit-order
Screen Shot 2021-11-10 at 2 26 10 PM

File: src/app_service/service/ens/service.nim [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  return none(string)
+
+method decodeENSContentHash*(self: Service, value: string): tuple[ensType: ENSType, output: string] =
+  if value == "":
+    return (ENSType.UNKNOWN, "")
+
+  if value[0..5] == "e40101":
+    return (ENSType.SWARM, value.split("1b20")[1])
+
+  if value[0..7] == "e3010170":
```
Comment (jrainville) [inline src/app_service/service/ens/service.nim:43]: There are a couple of string constants here. Is it possible to put them is `const`s?

File: src/app_service/service/provider/service.nim (hunk 1 above)
Comment (jrainville) [reply src/app_service/service/provider/service.nim:111]: Unrelated, but this function is pretty big. Maybe we could split it in sub-functions for each rpcMethod.

**Review by jrainville — COMMENTED** [review]
Nice job. Just 2 minor comments

**Comment by anastasiyaig** [comment]
@richard-ramos 

can we make the UI more readable? This screenshot is from current branch. The issues are with the label of wallet (the grey background overlaps the modal) and the text is not centred (and should be in several lines to make it possible to read) and drop-down looks odd

Screenshot 2021-11-16 at 15 32 07

Screenshot 2021-11-16 at 15 36 57


this one is from Beta 0.3.0

Screenshot 2021-11-16 at 15 34 59

**Review by borismelnik — APPROVED** [review]

**Comment by richard-ramos** [comment]
> can we make the UI more readable?

Done!

**Comment by anastasiyaig** [comment]
it is better now, thanks @richard-ramos :)

**Comment by anastasiyaig** [comment]
Screenshot 2021-11-19 at 11 24 40


Screenshot 2021-11-19 at 11 25 12

**Review by iurimatias — APPROVED** [review]

**Pull request merged**